sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/models/ScoringParameter.java

```diff
@@ -0,0 +1,110 @@
+import java.util.stream.Collectors;
+
+/**
+ * Represents a parameter value to be used in scoring functions (for example, referencePointParameter).
+ */
+public final class ScoringParameter {
+    private final ClientLogger logger = new ClientLogger(ScoringParameter.class);
+    private final String name;
+    private final List<String> values;
+
+    private static final String SEPARATOR = "-";
+    private static final String COMMA = ",";
+    private static final String SINGLE_QUOTE = "'";
+    /**
+     * Initializes a new instance of the ScoringParameter class with the given name and string values.
+     *
+     * @param name Name of the scoring parameter.
+     * @param values Values of the scoring parameter.
+     * @throws NullPointerException if {@code name} or {@code values} is null.
+     */
+    @JsonCreator
+    public ScoringParameter(@JsonProperty(value = "name") String name,
+        @JsonProperty(value = "values") List<String> values) {
+        Objects.requireNonNull(name);
+        Objects.requireNonNull(values);
+        this.name = name;
+        this.values = new ArrayList<>();
+        this.values.addAll(values);
+    }
+
+    /**
+     * Initializes a new instance of the ScoringParameter class with the given name and GeographyPoint value.
+     *
+     * @param name Name of the scoring parameter.
+     * @param value Value of the scoring parameter.
+     */
+    public ScoringParameter(String name, GeoPoint value) {
+        this(name, toLonLatStrings(value));
+    }
+
+    /**
+     * Gets the name of the scoring parameter.
+     *
+     * @return The name of scoring parameter.
+     */
+    public String getName() {
+        return name;
+    }
+
+    /**
+     * Gets the values of the scoring parameter.
+     *
+     * @return The values of scoring parameter.
+     */
+    public List<String> getValues() {
+        return values;
+    }
+
+    private static List<String> toLonLatStrings(GeoPoint point) {
+        if (point == null) {
+            return null;
+        }
+        return Arrays.asList(String.valueOf(point.getLongitude()), String.valueOf(point.getLatitude()));
+    }
+
+    /**
+     * Covert {@link ScoringParameter} to string.
+     *
+     * @return Service accepted string format.
+     * @throws IllegalArgumentException if all values in the list are null or empty.
+     */
+    @Override
+    @JsonValue
+    public String toString() {
+        String flattenValue = values.stream().filter(value -> !CoreUtils.isNullOrEmpty(value))
+            .map(this::escapeValue).collect(Collectors.joining(COMMA));
+        if (CoreUtils.isNullOrEmpty(flattenValue)) {
+            throw logger.logExceptionAsError(
+                new IllegalArgumentException("There must be at least one valid value for scoring parameter values."));
+        }
+        return name + SEPARATOR + values.stream().filter(value -> !CoreUtils.isNullOrEmpty(value))
+                .map(this::escapeValue).collect(Collectors.joining(COMMA));
```

Comment:
Why do we need to repeat creating `flattenValue`? This could be simplified to the following:

```java
return name + SEPARATOR + flattenValue;
```